x/gov/handler.go

```diff
@@ -91,17 +80,17 @@ func (h VoteHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (
 		return nil, err
 	}
 
-	switch oldVote, err := h.voteBucket.GetVote(db, voteMsg.ProposalID, vote.Elector.Address); {
-	case errors.ErrNotFound.Is(err): // not voted before: skip undo
-	case err != nil:
-		return nil, errors.Wrap(err, "failed to load vote")
-	default:
-		if err := proposal.UndoCountVote(*oldVote); err != nil {
+	oldVote, err := h.voteBucket.GetVote(db, voteMsg.ProposalID, vote.Elector.Address)
+	if !errors.ErrNotFound.Is(err) { // just skip on NotFound
```

Comment:
Cuz tests fail otherwise!

If I get a real error, I fail.
If I find an old vote, I need to "uncount" it.
If there is no old vote, do nothing.

Can you propose a better comment text?